File: Python/sum_square_even_root_odd.py
import math

def sum_square_even_root_odd(nums):
    lenList = len(nums)
    count = 0
    result = 0

    while count < lenList:
        if nums[count] % 2 == 0:
            result += nums[count] * nums[count]
            count += 1
        else:
            result += math.sqrt(nums[count])
            count += 1


    return round(result, 2)


print(sum_square_even_root_odd([4, 5, 7, 8, 1, 2, 3, 0]))

File: JavaScript/SumSquareEvenRootOdd.js
const sumSquareEvenRootOdd = (array) => {
  const lenArray = array.length;
  let count = 0;
  let result = 0;

  while (count < lenArray) {
    if (array[count] % 2 === 0) {
      result += array[count] * array[count]
      count += 1;
    } else {
      result += Math.sqrt(array[count]);
      count += 1;
    };
  };

  return +result.toFixed(2);
};

console.assert(sumSquareEvenRootOdd([4,5,7,8,1,2,3,0]) === 91.61, 'Erro no teste 1')
// console.log("====================================================")
// console.assert(sumSquareEvenRootOdd([1,14,9,8,17,21]) === 272.71, 'Erro no teste 2')